src/content/docs/en/guides/content-collections.mdx

```diff
@@ -543,6 +543,48 @@ If you have an existing Astro project, such as a blog, that uses Markdown or MDX
  
 See how to convert a basic blog example from `src/pages/posts/` to `src/content/posts` in our [step-by-step tutorial](/en/tutorials/add-content-collections/) that uses the codebase from [the Build a Blog tutorial's finished project](https://github.com/withastro/blog-tutorial-demo).
 
+## Enabling JSON Schema Generation
+
+<p><Since v="4.12.0" /></p>
+
+If you are working with data collections, you may wish to generate JSON Schema files to get better IntelliSense and type-checking in your editor. Astro provides a flag to auto-generate a JSON Schema for content collections of `type: 'data'` which can be used as the `$schema` value. This feature generates JSON Schema files for each data collection in your project, using a library called [zod-to-json-schema](https://github.com/StefanTerdell/zod-to-json-schema#known-issues)
```

Comment:
Normally when we use the word "better", we want to say what we're comparing something to.
Is the point that a JSON schema is better for a data collection than having the schema defined in the `.ts` file like a content collection? Is the *only* purpose for this to improve the editor experience?

Reply:
The JSON schema is automatically generated, based on the users defined schema.

"better" in terms of DX inside of the editor.

In theory there can be more tools users can add to use that JSON schema for other purposes, but I think 99% of users, use it to improve their DX and editor experience.

Reply:
Yes, but you still didn't tell me better than WHAT?? I don't know what this is better than, or what the other option is.  :smile:

Reply:
Better than not having autocompletion and red squiggle lines in the editor 😅

Reply:
Are you telling me that just having the default schema for data collections currently provides no intellisense? This is what I'm trying to figure out, what currently happens. Data collections are not currently type safe? :smile:

Reply:
It will not show you the intellisense or any type errors in the editor today. _(It will only show you the type errors in the browser)_

Reply:
Interesting, OK! I didn't know that was a current limitation of data collections! Thanks!